src/main/kotlin/com/angels/challenge/controller/TestController.kt

```diff
@@ -0,0 +1,68 @@
+package com.angels.challenge.controller
+
+import com.angels.challenge.repository.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.hateoas.MediaTypes
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+
+@RestController
+class TestController() {
+    @Autowired
+    var repository: ClusterRepository? = null
+    @Autowired
+    var repositoryRegions: RegionRepository? = null
+    @Autowired
+    var storeRepository: StoreRepository? = null
+    @Autowired
+    var productRepository: ProductRepository? = null
+    @Autowired
+    var seasonRepository: SeasonRepository? = null
+    @Autowired
+    var storeSeasonRepository: StoreSeasonRepository? = null
+
+    @GetMapping("/clusters", produces = [MediaTypes.HAL_JSON_VALUE, MediaType.APPLICATION_JSON_VALUE])
+    @ResponseBody
+    fun getClusters(): ResponseEntity<Any?> {
+        val list = repository?.findAll()
+        return ResponseEntity(list, HttpStatus.OK)
+    }
+
+    @GetMapping("/regions", produces = [MediaTypes.HAL_JSON_VALUE, MediaType.APPLICATION_JSON_VALUE])
+    @ResponseBody
+    fun getRegions(): ResponseEntity<Any?> {
+        val list = repositoryRegions?.findAll()
+        return ResponseEntity(list, HttpStatus.OK)
+    }
+
+    @GetMapping("/stores", produces = [MediaTypes.HAL_JSON_VALUE, MediaType.APPLICATION_JSON_VALUE])
+    @ResponseBody
+    fun getStores(): ResponseEntity<Any?> {
+        val list = storeRepository?.findAll()
+        return ResponseEntity(list, HttpStatus.OK)
+    }
+
+    @GetMapping("/products", produces = [MediaTypes.HAL_JSON_VALUE, MediaType.APPLICATION_JSON_VALUE])
+    @ResponseBody
+    fun getProducts(): ResponseEntity<Any?> {
+        val list = productRepository?.findAll()
```

Comment:
The controller layer is making use of persistence classes directly. This means that controllers and persistence are tightly coupled. What are the consequences of this design decision and what could you do to mitigate these consequences?

Reply:
I believe that it would better to include a service layer to communicate with the persistance classes and the controller instead using the service layer. A possible consequence would be if we wanted to change the database then we would need to refactor the all the controlers coupled with it, instead of just chaging the service responsible for communicating with the persistance layer

Reply:
👍